bliss/simulator/background.py

```diff
@@ -1,67 +1,36 @@
-from typing import Dict, Tuple
+from typing import Tuple
 
 import numpy as np
 import torch
-from einops import rearrange
-from omegaconf import OmegaConf
-from omegaconf.dictconfig import DictConfig
 from torch import Tensor, nn
 
-# ideally this module wouldn't depend on any specific survey
-from bliss.surveys.sdss import SloanDigitalSkySurvey
 
-
-class ConstantBackground(nn.Module):
-    def __init__(self, background: Tuple[float, ...], **_kwargs):
-        super().__init__()
-        bg: Tensor = torch.tensor([background])
-        bg = rearrange(bg[0], "c -> 1 c 1 1")
-        self.register_buffer("background", bg, persistent=False)
-
-    def sample(self, shape, **_kwargs) -> Tensor:
-        assert isinstance(self.background, Tensor)
-        batch_size, c, hlen, wlen = shape
-        return self.background.expand(batch_size, c, hlen, wlen)
-
-
-class SimulatedSDSSBackground(nn.Module):
-    def __init__(self, sdss_fields: DictConfig):
+class ImageBackground(nn.Module):
+    def __init__(
+        self,
+        items,
+        bands: Tuple[int, ...] = (0, 1, 2, 3, 4),
```

Comment:
Imo bands should be a required parameter instead of having a default value, to ensure it has to be specified in the survey.

Reply:
Done.